src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -57,117 +53,65 @@ const EmbeddingPreview = () => {
-      dataGenerator: generateDataCategoricalSpec
       imgSrc: plot1Pic,
       plotUuid: 'embeddingPreviewBySample',
       plotType: 'embeddingCategorical',
     },
     cellCluster: {
       title: 'Colored by CellSets',
-      specGenerator: generateEmbeddingCategoricalSpec,
-      dataGenerator: generateDataCategoricalSpec
       imgSrc: plot1Pic,
       plotUuid: 'embeddingPreviewByCellSets',
       plotType: 'embeddingCategorical',
     },
     mitochondrialFraction: {
       title: 'Mitochondrial fraction reads',
-      specGenerator: generateEmbeddingContinuousSpec,
-      dataGenerator:
-        imgSrc: plot2Pic,
+      imgSrc: plot2Pic,
       plotUuid: 'embeddingPreviewMitochondrialReads',
       plotType: 'embeddingContinuous',
     },
     doubletScore: {
       title: 'Cell doublet score',
-      specGenerator: generateEmbeddingContinuousSpec,
       imgSrc: plot2Pic,
       plotUuid: 'embeddingPreviewDoubletScore',
       plotType: 'embeddingContinuous',
     },
   };
 
-  const config = useSelector((state) => state.componentConfig[plots[selectedSpec].plotUuid]?.config);
-
-  // Prepare data for categorical embedding
-  const embeddingType = embeddingConfig.configureEmbedding?.embeddingSettings.method;
-
-  const cellSets = useSelector((state) => state.cellSets);
-  const { data: embeddingData } = useSelector((state) => state.embeddings[embeddingType]) || {};
-
-  useEffect(() => {
-    if (!embeddingData) {
-      dispatch(loadEmbedding(experimentId, embeddingType));
-      dispatch(loadCellSets(experimentId));
-    }
-  }, [experimentId]);
+  const config = useSelector((state) => state.componentConfig[plots[selectedPlot].plotUuid]?.config);
 
   useEffect(() => {
     // Do not update anything if the cell sets are stil loading or if
     // the config does not exist yet.
-    if (!config || !embeddingData) {
-      return;
-    }
-
-    if (config.shownGene) {
-      dispatch(loadGeneExpression(experimentId, [config.shownGene]));
-    }
-
-    if ((plots[selectedSpec].plotType === 'embeddingContinuous') || !cellSets) {
+    if (!config) {
       return;
     }
 
-    const spec = plots[selectedSpec].specGenerator(config);
+    switch (plots[selectedPlot].plotType) {
+      case 'embeddingContinuous':
+        setPlot(<ContinuousEmbeddingPlot experimentId={experimentId} config={config} />);
+        break;
 
-    if (plots[selectedSpec].plotType === 'embeddingContinuous') {
-      generateContinuousSpecData(spec, config.shownGene, embeddingData)
-    } else {
-      generateCategoricalSpecData(spec, cellSets, config.selectedCellSet, embeddingData)
+      // Set this as default plot because default plot is 'sample' which is categorical
+      case 'embeddingCategorical':
+      default:
+        setPlot(<CategoricalEmbeddingPlot experimentId={experimentId} config={config} />);
+        break;
     }
-    setPlotSpec(spec);
-
-  }, [config, embeddingData, selectedExpression, cellSets]);
+  }, [config]);
```

Comment:
This refactors all the code to generate categorical and continuous embedding plot into its own components, committed together in 075d9499